db/memtable_list.cc

```diff
@@ -361,6 +371,15 @@ void MemTableList::PickMemtablesToFlush(uint64_t max_memtable_id,
   if (!atomic_flush || num_flush_not_started_ == 0) {
     flush_requested_ = false;  // start-flush request is complete
   }
+
+  // Sort the list of memtables by increasing memtable ID.
+  // This is useful when the mempurge feature is activated
+  // and the memtables are not guaranteed to be sorted in
+  // the memlist vector.
+  std::sort(ret->begin(), ret->end(),
```

Comment:
Based on caller, I believe this happens under DB mutex. Probably not a big deal, but noting that in case it can be delayed or something. Or should we keep the source list sorted?

Reply:
I'm torn on this because keeping the source list sorted would also imply iterating through the source list with the DB mutex. Overall this `ret` array is extremely small in practice (no way there is more than 10 pointers in it at a time... more than 2 would honestly be surprising), so std::sort would be lightning fast.
Another option would be to use another data structure to store the immutable memtables (to use an ordered set for example), but I feel like this is a much more invasive change and that change may bring its own set of challenges.